src/test/test-resize-fs.c

```diff
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <linux/magic.h>
+
+#include "missing_magic.h"
+#include "resize-fs.h"
+#include "tests.h"
+
+TEST(resize_fs) {
+        uint64_t *k = NULL;
+        assert_se (resize_fs(2, UINT64_MAX, k) == -ERANGE);
+        assert_se (resize_fs(2, 0, k) == -ERANGE);
+}
+
+TEST(minimal_size_by_name) {
+        assert_se (minimal_size_by_fs_name("ext4") == EXT4_MINIMAL_SIZE);
+        assert_se (minimal_size_by_fs_name("xfs") == XFS_MINIMAL_SIZE);
+        assert_se (minimal_size_by_fs_name("btrfs") == BTRFS_MINIMAL_SIZE);
+        assert_se (minimal_size_by_fs_name("others") == UINT64_MAX);
+}
+
+TEST(minimal_size_by_type) {
+        assert_se (minimal_size_by_fs_magic(EXT4_SUPER_MAGIC) == EXT4_MINIMAL_SIZE);
+        assert_se (minimal_size_by_fs_magic(EXT4_SUPER_MAGIC) == EXT4_MINIMAL_SIZE);
```

Comment:
> same line twice

Oh..I meant to use the second line for XFS sir.

Reply:
The major issue I've had with this test is creating a file descriptor of the different filesystem types: XFS, EXT4, and BTRFS.

@bluca said I needed to test the resize-fs function as well as all it's helper functions.

I've been able to test the helper functions, but to test the main resize-fs() function, I need to pass in fds of various filesystem types.

@DaanDeMeyer told me to use the make_filesystem() function to create files of different types, and grow the size of the underlying file with ftruncate.

I spent several hours last Sunday trying to get it to work, but make_filesystem just wouldn't work.


So I decided to write a test for a new function altogether, since I still have some other tasks to complete.

I'm glad I left this PR open though, cos I learnt about the NULL pointer today from @poettering .


Maybe if it stays open for a little longer, I could learn enough to get it to work.


Thank you all sirs.